File: frontend/tennis-time/libs/reservation/src/lib/reservation/realtime/slot-status-notification.ts
import { ReservationStatus, SlotBasketState } from '../types';

export type SlotBasketStateNotification = Partial<SlotBasketState>;

export interface SlotStatusNotification {
  slotId: string;
  compositeSlotId: string;
  serviceId: string;
  status: ReservationStatus;
  reservationDate?: string | null;
  userId?: string | null;
  basketState?: SlotBasketStateNotification | null;
}

File: bff/src/main/java/com/tennistime/bff/domain/realtime/SlotStatusNotification.java
package com.tennistime.bff.domain.realtime;

import com.tennistime.bff.domain.model.types.ReservationStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Data structure carrying slot status information received from upstream services.
 */
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SlotStatusNotification {

    private String slotId;
    private String compositeSlotId;
    private String serviceId;
    private ReservationStatus status;
    private String reservationDate;
    private String userId;
    private SlotBasketStateNotification basketState;
}